### DISCUSSION TITLE
Concatenate several worksheets into one.

### DISCUSSION THREAD

**Discussion by Andrei-Stefan20** [PHPOffice/PhpSpreadsheet#3203] [Q&A]
How can I concatenate several worksheets into one? I would need to export the data as CSV.
Are there alternative ways to save the different worksheets in one CSV file?

**Comment by MarkBaker** [comment]
The only practical option to do this is to write each worksheet to a separate file, and then to concatenate those files together; PhpSpreadsheet doesn't provide an option to write all worksheets to a single Csv file.

**Reply by Andrei-Stefan20** [reply]
Thanks for the reply.

I currently have several spreadsheets where I extract individual worksheets and put them into a new spreadsheet that has all worksheets.
Is there a way to extract each worksheet's data into a worksheet? For example by concatenating tables by separating them from empty rows.

I rephrase the question.
Is there a way to manipulate the data inside the worksheets to concatenate them?

**Reply by MarkBaker** [reply]
There's no built-in method, no equivalent to Excel's "Consolidate" that will merge data from one worksheet to another; you could use `toArray()` and `fromArray()` to copy the raw values, but that won't copy styles and could break formulae.

**Reply by Andrei-Stefan20** [reply]
Thanks again for the support, I think I'm going to do a classic merge of separate CSV files.
